### PR TITLE
Adapt code to latest GEMOC framework refactorings

### PR DESCRIPTION
**Pull request by ebousse**

### CODE REVIEW

File: uk.ac.kcl.inf.modelling.xdsml.gemoc_henshin/src/uk/ac/kcl/inf/modelling/xdsml/gemoc_henshin/engine/core/HenshinConcurrentExecutionEngine.xtend [hunk 1]
```diff
@@ -260,5 +258,13 @@ class HenshinConcurrentExecutionEngine extends AbstractConcurrentExecutionEngine
 			pcs.addPropertyChangeListener("semantics", pcl)
 		}
 
+		override getEngine() {
+			return engine
+		}
+
+		override getModelRoot() {
+			return engine.executionContext.resourceModel.contents.get(0)
```
Comment (szschaler) [inline uk.ac.kcl.inf.modelling.xdsml.gemoc_henshin/src/uk/ac/kcl/inf/modelling/xdsml/gemoc_henshin/engine/core/HenshinConcurrentExecutionEngine.xtend:266]: Use Xtend notation here.

**Review by szschaler — CHANGES_REQUESTED** [review]
Minor changes, plus need to check some things in running system. @szschaler will pick these up.

**Comment by szschaler** [comment]
Thanks, @ebousse for adding this in already. I want to check it in GEMOC briefly, and will merge as soon as I have done that. Otherwise, this looks good to me!

**Comment by szschaler** [comment]
Meh. While this now compiles against GEMOC Nightly Build, it doesn't actually appear to work. There appear to be two issues:

1. [ ] A module export issue with Henshin 1.8 which means that CPA doesn't work at present. I will need to check with Daniel to see what's going on.
2. [ ] An issue within GEMOC/the Henshin engine that means that while steps are calculated, they cannot currently be executed. Clicking on a step in the step selector freezes the engine before even asking it to execute a single step. (In my current configuration, I also don't see the the omniscient debugger interface, but not sure if this is related...)

@ebousse, it would be good to look over the second issue together if you're available. Probably easiest to switch to email initially...

**Comment by ebousse** [comment]
> An issue within GEMOC/the Henshin engine that means that while steps are calculated, they cannot currently be executed. Clicking on a step in the step selector freezes the engine before even asking it to execute a single step. (In my current configuration, I also don't see the the omniscient debugger interface, but not sure if this is related...)

I did try the code before submitting the PR, and could run a model without problem, so that's strange.

Just a hinch: can you check that, in the launch configuration, the checkbox "break at start" is _not_ ticked?

**Comment by ebousse** [comment]
There is indeed a problem with the timeline view though, I'll look into that

**Comment by szschaler** [comment]
Removing the "Break at start" check box fixed the problem at least for when running in debug mode. If I try to run a model in run mode, then I get errors about inability to write model state. I guess these are broader GEMOC issues, though (as is the timeline view thing, so for now propose to merge here.

**Review by szschaler — APPROVED** [review]

**Pull request merged**